src/Tickets/Hooks.php

```diff
@@ -48,6 +49,103 @@ public function register() {
 	 */
 	protected function add_actions() {
 		$this->container->register( Ticket_Cache_Controller::class );
+
+		add_action( 'admin_post_tec_tickets_remove_orphans', [ $this, 'remove_orphans' ] );
+		add_action( 'tec_tickets_remove_orphans_action', [ $this, 'remove_orphans_action' ], 10, 1 );
+
+		// Cache invalidation hooks for orphaned posts.
+		add_action( 'deleted_post', [ $this, 'clear_orphaned_posts_cache' ] );
+		add_action( 'wp_trash_post', [ $this, 'clear_orphaned_posts_cache' ] );
+		add_action( 'untrashed_post', [ $this, 'clear_orphaned_posts_cache' ] );
+	}
+
+	/**
+	 * Trash orphaned entries.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function remove_orphans() {
+		// Bail if not admin.
+		if ( ! current_user_can( 'manage_options' ) ) {
+			wp_die( esc_html_x( 'Not allowed.', 'Orphaned posts', 'event-tickets' ) );
+		}
+
+		// Bail if nonce verification fails.
+		$nonce = tec_get_request_var( 'nonce' );
+		if ( ! wp_verify_nonce( $nonce, 'tec_tickets_remove_orphans' ) ) {
+			wp_die( esc_html_x( 'Nonce verification failed.', 'Orphaned posts', 'event-tickets' ) );
+		}
+
+		// Bail if no provider specified.
+		$provider = tec_get_request_var( 'provider', false );
+		if ( ! $provider ) {
+			wp_die( esc_html_x( 'No provider specified.', 'Orphaned posts', 'event-tickets' ) );
+		}
+
+		// Get IDs.
+		if ( $provider === 'rsvp' ) {
+			$ids = tribe( Tribe__Tickets__RSVP::class )->get_orphaned_posts( false );
+		} elseif ( $provider === 'tc_ticket' ) {
+			$ids = tribe( \TEC\Tickets\Commerce\Module::class )->get_orphaned_posts( false );
+		} else {
+			wp_die( esc_html_x( 'Unsupported provider.', 'Orphaned posts', 'event-tickets' ) );
+		}
+
+		// Return URL.
+		$url = add_query_arg( 'page', 'tec-tickets-settings', wp_get_referer() );
```

Comment:
Since now we are using wp_get_referer make sure that the `page` param is not added twice after being redirected back :D